jupyter_events/logger.py

```diff
@@ -107,94 +79,48 @@ def get_handlers():
-
-        for p, attrs in schema["properties"].items():
-            if p.startswith("__"):
-                raise ValueError(
-                    "Schema {} has properties beginning with __, which is not allowed"
-                )
-
-            # Validate "categories" property in proposed schema.
-            try:
-                cats = attrs["categories"]
-                # Categories must be a list.
-                if not isinstance(cats, list):
-                    raise ValueError(
-                        'The "categories" field in a registered schemas must be a list.'
-                    )
-            except KeyError:
-                raise KeyError(
-                    'All properties must have a "categories" field that describes '
-                    'the type of data being collected. The "{}" property does not '
-                    "have a category field.".format(p)
-                )
-
-        self.schemas[(schema["$id"], schema["version"])] = schema
+        self.schemas.register_from_file(schema_file)
 
-    def get_allowed_properties(self, schema_name):
-        """Get the allowed properties for an allowed schema."""
-        config = self.allowed_schemas[schema_name]
-        try:
-            return set(config["allowed_properties"])
-        except KeyError:
-            return set()
+    def register_handler(self, handler: logging.Handler):
+        """Register a new logging handler to the Event Logger.
 
-    def get_allowed_categories(self, schema_name):
+        All outgoing messages will be formatted as a JSON string.
         """
-        Return a set of allowed categories for a given schema
-        from the EventLog's config.
-        """
-        config = self.allowed_schemas[schema_name]
-        try:
-            allowed_categories = config["allowed_categories"]
-            allowed_categories.append("unrestricted")
-            return set(allowed_categories)
-        except KeyError:
-            return {"unrestricted"}
 
-    def record_event(self, schema_name, version, event, timestamp_override=None):
+        def _skip_message(record, **kwargs):
+            """
+            Remove 'message' from log record.
+            It is always emitted with 'null', and we do not want it,
+            since we are always emitting events only
+            """
+            del record["message"]
+            return json.dumps(record, **kwargs)
+
+        formatter = jsonlogger.JsonFormatter(json_serializer=_skip_message)
+        handler.setFormatter(formatter)
+        self.log.addHandler(handler)
+        if handler not in self.handlers:
+            self.handlers.append(handler)
+
+    def remove_handler(self, handler: logging.Handler):
+        """Remove a logging handler from the logger and list of handlers."""
+        self.log.removeHandler(handler)
+        if handler in self.handlers:
+            self.handlers.remove(handler)
+
+    def emit(self, id: str, version: int, data: dict, timestamp_override=None):
```

Comment:
Should `id` be `schema_name`?  There's a doc-string mismatch.  Personally, I find `id` to often be ambiguous, particularly when it doesn't equate to an actual "ID".

I see (looking below).  This is really equating to the `$id` field - thus `id`.  My preference is still `schema_name`, but now see the link between the two.

Reply:
I renamed this argument to `schema_id`. I think this removes some of the ambiguity from `id`, but remains true to the value it represents in the the schema itself, that is `$id`.